packages/docusaurus-plugin-sitemap/src/__tests__/createSitemap.test.ts

```diff
@@ -146,13 +146,32 @@ describe('createSitemap', () => {
     expect(sitemap).toContain('<loc>https://example.com/nested/test2</loc>');
   });
 
+  it('site with nested pages in routes', async () => {
+    const sitemap = await createSitemap(
+      {
+        url: 'https://example.com',
+      } as DocusaurusConfig,
+      [{ path: '/', routes: [{ path: '/test', routes: [{ path: '/test/nested' }] }] }],
+      {},
+      {
+        changefreq: EnumChangefreq.DAILY,
+        priority: 0.7,
+        ignorePatterns: [],
+        filename: 'sitemap.xml',
+      },
+    );
+    expect(sitemap).toContain('<loc>https://example.com/</loc>');
+    expect(sitemap).toContain('<loc>https://example.com/test</loc>');
+    expect(sitemap).toContain('<loc>https://example.com/test/nested</loc>');
+  });
+
```

Comment:
In this case, the https://example.com/test is not included, on purpose.

The fact that there's a parent layout being applied for all `/test/*` subroutes does not mean that any meaningful page exists at this address. This is generally the case because we usually enable a subroute to render at `/test/`  but it's not mandatory so we can't make this generalization.

One concrete example where this would cause troubles:

https://tutorial.docusaurus.io/docs/

This page is a valid nested route path, and yet it's a 404. Your PR would include it in the sitemap, while we definitively don't want to include it.

Reply:
We have pages missing from our sitemap. These pages are generated from the `@docusaurus/plugin-content-docs`. These pages are are of the form:

* `/docs/01-toplevel/index.md` - included in the sitemap as `/docs/toplevel` (obviously the leading `01` is striped at build time)
* `/docs/01-toplevel/02-nested/index.md` - not included in the sitemap

This isn't a layout, the 01-toplevel is a real page with real child pages. I'm guessing this works on the docusaurus site because the `slug` is being manually specified on every page.

Does what I'm saying make sense, or do I need to make a reproduction?

Reply:
Thanks
I don't understand how a non-layout doc could be missing.
Yes a reproduction is needed so that I can see the issue in practice on a concrete use-case.